common4j/src/main/com/microsoft/identity/common/java/util/CharArrayJsonAdapter.java

```diff
@@ -0,0 +1,19 @@
+package com.microsoft.identity.common.java.util;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+import java.io.IOException;
+
+public class CharArrayJsonAdapter extends TypeAdapter<char[]> {
```

Comment:
Javadoc

Reply:
Why is this class needed?

Reply:
CharArrayJsonAdapter is used with GSON to serialise character array as string and vice-versa.  The default implementation in GSON will serialise a character array to array of characters. In this case, we want passwords to be serialized as strings.

Reply:
Can you point me to where this serialization is being used?

Reply:
It is used in SignInTokenRequest.kt file line no 166